src/components/DataViewer/index.tsx

```diff
@@ -18,63 +18,87 @@ import {
   clearDataset,
   loadDataset,
   updateAdminId,
+  AdminBoundaryParams,
+  EWSParams,
+  loadEWSDataset,
 } from '../../context/datasetStateSlice';
 import { dateRangeSelector } from '../../context/mapStateSlice/selectors';
 import Chart from '../DataDrawer/Chart';
 import { ChartConfig } from '../../config/types';
 import { useSafeTranslation } from '../../i18n';
 
+const isAdminBoundary = (
+  params: AdminBoundaryParams | EWSParams,
+): params is AdminBoundaryParams =>
+  (params as AdminBoundaryParams).id !== undefined;
+
 function DataViewer({ classes }: DatasetProps) {
   const dispatch = useDispatch();
   const isDatasetLoading = useSelector(loadingDatasetSelector);
   const { startDate: selectedDate } = useSelector(dateRangeSelector);
   const { t } = useSafeTranslation();
 
-  const { data: dataset, adminBoundaryParams: params, id } = useSelector(
-    datasetSelector,
-  );
+  const {
+    data: dataset,
+    datasetParams: params,
+    title,
+    chartType,
+  } = useSelector(datasetSelector);
 
   useEffect(() => {
-    if (params && selectedDate && id) {
+    if (!params || !selectedDate) {
+      return;
+    }
+
+    if (isAdminBoundary(params)) {
+      const { id, boundaryProps, url, serverLayerName } = params;
       dispatch(
         loadDataset({
```

Comment:
I think some of the logic should be moved into `loadDataset` and `loadDataset` should be the one function that behaves differently based on the `loader`. wdyt?